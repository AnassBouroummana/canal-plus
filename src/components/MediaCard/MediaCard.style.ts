import { Card, Typography } from '@mui/material';
import styled from 'styled-components';
import { getSpacing } from 'stylesheet';

export const StyledCard = styled(Card)`
  && {
    margin: ${getSpacing(2)};
    max-width: 345px;
    display: flex;
    flex: 1 0 21%;
    flex-direction: column;
  }
`;
StyledCard.displayName = 'StyledCard';

export const StyledOverview = styled(Typography)`
  && {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
`;
StyledOverview.displayName = 'StyledOverview';
