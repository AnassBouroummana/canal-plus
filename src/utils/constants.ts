export const NODE_ENV = process.env.NODE_ENV || '';
export const IMG_BASE_PATH = 'https://image.tmdb.org/t/p/original/';
export const BASE_PATH = 'https://api.themoviedb.org/3/';
export const API_KEY = '92b418e837b833be308bbfb1fb2aca1e';
