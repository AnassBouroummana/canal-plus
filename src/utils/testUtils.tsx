import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Router } from 'react-router';
import { createBrowserHistory } from 'history';
import ConnectedIntlProvider from 'providers/ConnectedIntlProvider';
import theme from 'stylesheet';

export const AllTheProviders: React.FC<{}> = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <Router history={createBrowserHistory()}>
        <ConnectedIntlProvider>{children}</ConnectedIntlProvider>
      </Router>
    </ThemeProvider>
  );
};

const customRender = (
  ui: Parameters<typeof render>[0],
  options?: Parameters<typeof render>[1],
) => ({
  ...render(ui, { ...options, wrapper: AllTheProviders }),
  history: createBrowserHistory(),
});

const testSnapshot = (comp: React.ReactElement) => {
  const { container } = customRender(comp);
  expect(container).toMatchSnapshot();
};

export * from '@testing-library/react';
export { customRender as render, testSnapshot };
