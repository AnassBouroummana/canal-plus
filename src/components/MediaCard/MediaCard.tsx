import * as React from 'react';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';
import { StyledCard, StyledOverview } from './MediaCard.style';
import { Media } from 'modules/Media';
import { IMG_BASE_PATH } from 'utils/constants';
import { useHistory } from 'react-router';

interface Props {
  media: Media;
}

const getMediaImagePath = (backdrop_path: string, poster_path: string) => {
  return IMG_BASE_PATH.concat(!!poster_path ? poster_path : backdrop_path);
};

const MediaCard: React.FC<Props> = ({ media }) => {
  const { id, name, overview, backdrop_path, poster_path } = media;
  const mediapath = getMediaImagePath(backdrop_path, poster_path);
  const history = useHistory();
  return (
    <StyledCard onClick={() => history.push(`/movies/${id}`)}>
      <CardMedia component="img" alt="poster image" height="200" image={mediapath} />
      <CardContent>
        <Typography gutterBottom variant="h5" component="div">
          {name}
        </Typography>
        <StyledOverview title={overview} variant="body2" color="text.secondary">
          {overview}
        </StyledOverview>
      </CardContent>
    </StyledCard>
  );
};
export default MediaCard;
