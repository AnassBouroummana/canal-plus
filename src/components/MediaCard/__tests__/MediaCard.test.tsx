import { render } from 'utils/testUtils';
import MediaCard from '../MediaCard';

const media = {
  id: 1,
  backdrop_path: '/zGqBpfck5wVzS2iy6EsENMZw5kP.jpg',
  name: 'Walking the Americas',
  overview:
    'Explorer Levison Wood sets out to trek 1800 miles from Mexico to Colombia, through fascinating, beautiful and diverse regions, and meets people living everywhere from violent cities to deep jungle.',
  poster_path: '/6NUjtmU99inCbQ6TkwoTDSQuzIU.jpg',
};
describe('MediaCard', () => {
  it('test if has title and overview ', () => {
    const { getByText } = render(<MediaCard media={media} />);
    const title = getByText(media.name);
    const overview = getByText(media.overview);
    expect(title).toBeInTheDocument();
    expect(overview).toBeInTheDocument();
  });
  it('MediaCard:: must match the snapshot', () => {
    const { container } = render(<MediaCard media={media} />);
    expect(container).toMatchSnapshot();
  });
});
