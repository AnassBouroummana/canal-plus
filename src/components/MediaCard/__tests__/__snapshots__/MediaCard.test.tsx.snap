// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaCard MediaCard:: must match the snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-bdvvtL inSzaB css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="poster image"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="200"
      src="https://image.tmdb.org/t/p/original//6NUjtmU99inCbQ6TkwoTDSQuzIU.jpg"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        Walking the Americas
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 sc-gsDKAQ dfClLr css-r40f8v-MuiTypography-root"
        title="Explorer Levison Wood sets out to trek 1800 miles from Mexico to Colombia, through fascinating, beautiful and diverse regions, and meets people living everywhere from violent cities to deep jungle."
      >
        Explorer Levison Wood sets out to trek 1800 miles from Mexico to Colombia, through fascinating, beautiful and diverse regions, and meets people living everywhere from violent cities to deep jungle.
      </p>
    </div>
  </div>
</div>
`;
